src/main/java/org/tc/services/decision/DecisionServiceImpl.java

```diff
@@ -29,21 +26,21 @@
     public void makeDecision(Decision decision, Course course) {
         decision.setCourseForReview(course);
         decisionDao.create(decision);
-        courseService.setReviewDecision(course);
+        courseService.processReviewResult(course);
         mailSender.sendApprovalUpdate(decision);
     }
 
     @Override
     public Optional<Decision> getKnowledgeManagerDecision(Course course) {
         Optional<Decision> optionalDecision = course.getDecisions().stream().filter(decision ->
-                decision.getManager().getId() == roleService.getKnowledgeManager().getId()).findFirst();
+                decision.getManager().getId() == roleService.getKnowledgeManager().get().getId()).findFirst();
         return optionalDecision;
     }
 
     @Override
     public Optional<Decision> getDepartmentManagerDecision(Course course) {
         Optional<Decision> optionalDecision = course.getDecisions().stream().filter(decision ->
-                decision.getManager().getId() == roleService.getDepartmentManager().getId()).findFirst();
+                decision.getManager().getId() == roleService.getDepartmentManager().get().getId()).findFirst();
```

Comment:
I think you will have issues with NPE as main pattern for using Optional is

if (opt.isPresent()) {
  // do logic with opt.get()
}

Also Optional helps in streams to avoid nulls.